### PR TITLE
fix: modelmanager mock tests for model creation

### PR DESCRIPTION
**Pull request by Aflynn50**
Change the tests to have a non nil model service, this means the mocks calls now need to be expected for model create. Before, if the model service was nil the model was not created in DQLite so this didn't need to be tested.

This change is needed as part of #18003 

## Checklist

- [x] Code style: imports ordered, good names, simple structure, etc
- [x] Comments saying why design decisions were made
- [x] Go unit tests, with comments saying what you're testing

## QA steps
```
go test -v -tags=libsqlite3,dqlite ./apiserver/facades/client/modelmanager/... -check.v -check.f Suite
```

https://warthogs.atlassian.net/browse/JUJU-6626

### CODE REVIEW

File: apiserver/facades/client/modelmanager/modelinfo_test.go [hunk 1]
```diff
@@ -344,7 +346,7 @@ func (s *modelInfoSuite) expectedModelInfo(c *gc.C, credentialValidity *bool) pa
 func (s *modelInfoSuite) TestModelInfo(c *gc.C) {
 	api, ctrl := s.getAPI(c)
 	defer ctrl.Finish()
-	s.mockAccessService.EXPECT().GetModelUsers(gomock.Any(), user.AdminUserName, coremodel.UUID(s.st.model.cfg.UUID())).Return(s.modelUserInfo, nil)
+	s.mockAccessService.EXPECT().GetModelUsers(gomock.Any(), gomock.Any(), coremodel.UUID(s.st.model.cfg.UUID())).Return(s.modelUserInfo, nil)
```
Comment (barrettj12) [inline apiserver/facades/client/modelmanager/modelinfo_test.go:349]: What's the rationale for this change? Better to assert the username going in is correct if possible IMO?

File: apiserver/facades/client/modelmanager/modelmanager_test.go [hunk 2]
```diff
@@ -303,6 +308,77 @@ func (s *modelManagerSuite) setAPIUser(c *gc.C, user names.UserTag) {
...
+	networkService := mocks.NewMockNetworkService(ctrl)
+	modelConfigService := mocks.NewMockModelConfigService(ctrl)
+	modelServiceFactory.EXPECT().ModelInfo().Return(modelInfoService)
+	modelServiceFactory.EXPECT().Network().Return(networkService)
+	modelServiceFactory.EXPECT().Config().Return(modelConfigService)
+
+	// Expect calls to functions of the model services.
+	modelInfoService.EXPECT().CreateModel(gomock.Any(), s.controllerUUID)
+	modelConfigService.EXPECT().SetModelConfig(gomock.Any(), modelConfig)
+	networkService.EXPECT().ReloadSpaces(gomock.Any())
```
Comment (barrettj12) [inline apiserver/facades/client/modelmanager/modelmanager_test.go:379]: You might want to put `.AnyTimes()` after these expectations, otherwise you're asserting that they are called exactly once.

File: apiserver/facades/client/modelmanager/modelmanager_test.go [hunk 3]
```diff
@@ -482,15 +540,32 @@ func (s *modelManagerSuite) testCreateModelDefaultCredentialAdmin(c *gc.C, owner
...
+	bobUUID := usertesting.GenUserUUID(c)
+	modelUUID := modeltesting.GenModelUUID(c)
+
+	s.modelService.EXPECT().DefaultModelCloudNameAndCredential(
+		gomock.Any()).Return("dummy", credential.Key{}, nil)
+	s.accessService.EXPECT().GetUserByName(
+		gomock.Any(), bobName,
+	).Return(user.User{UUID: bobUUID}, nil)
+
+	s.modelService.EXPECT().CreateModel(gomock.Any(), model.ModelCreationArgs{
```
Comment (barrettj12) [inline apiserver/facades/client/modelmanager/modelmanager_test.go:556]: So why can't this one be rolled into the same `expectCreateModel` function?

File: apiserver/facades/client/modelmanager/modelmanager_test.go [hunk 4]
```diff
@@ -515,38 +590,27 @@ func (s *modelManagerSuite) TestCreateModelNoDefaultCredentialNonAdmin(c *gc.C)
 	c.Assert(err, gc.ErrorMatches, "no credential specified")
 }
 
-func (s *modelManagerSuite) TestCreateModelUnknownCredential(c *gc.C) {
```
Comment (barrettj12) [inline apiserver/facades/client/modelmanager/modelmanager_test.go:518]: Should we keep this test?

File: apiserver/facades/client/modelmanager/modelmanager_test.go [hunk 5]
```diff
@@ -1101,6 +1224,63 @@ func (s *modelManagerStateSuite) setAPIUser(c *gc.C, user names.UserTag) {
...
+		func(context.Context) error { return nil },
+		nil,
+	)
+
+	// Expect call to get the model service factory.
+	modelServiceFactory := mocks.NewMockModelServiceFactory(ctrl)
+	s.serviceFactoryGetter.EXPECT().ServiceFactoryForModel(gomock.Any()).Return(modelServiceFactory)
+
+	// Expect calls to get various model services.
+	modelInfoService := mocks.NewMockModelInfoService(ctrl)
```
Comment (barrettj12) [inline apiserver/facades/client/modelmanager/modelmanager_test.go:1267]: This all looks copy pasted from the other function no?

File: apiserver/facades/client/modelmanager/modelinfo_test.go (hunk 1 above)
Comment (barrettj12) [reply apiserver/facades/client/modelmanager/modelinfo_test.go:349]: There are a lot of other similar changes in this file

**Review by barrettj12 — COMMENTED** [review]
Looks pretty reasonable overall, just a few things were unclear to me.

File: apiserver/facades/client/modelmanager/modelinfo_test.go (hunk 1 above)
Comment (Aflynn50) [reply apiserver/facades/client/modelmanager/modelinfo_test.go:349]: Ah, yea, so this is beacuse I made the changes for the other PR then separated them out here to make it easier to review. In the end though, I'm not sure it actually makes things better, I think I might just close this PR and say that this is just a commit in #18003.

A lot of the changes in here will be changed again with the changes to GetModelUsers in there.

File: apiserver/facades/client/modelmanager/modelmanager_test.go (hunk 2 above)
Comment (Aflynn50) [reply apiserver/facades/client/modelmanager/modelmanager_test.go:379]: For these I think its fine to expect exactly once. In create model they should all only be called once, if a retry is added or something then this can be updated in the tests

File: apiserver/facades/client/modelmanager/modelmanager_test.go (hunk 3 above)
Comment (Aflynn50) [reply apiserver/facades/client/modelmanager/modelmanager_test.go:556]: (I'm going to make these changes in the other PR and close this one.)
One calls the last method slightly differently because its not an admin user. I would have had to add more parameters to the expect function and didn't really want to here.

File: apiserver/facades/client/modelmanager/modelmanager_test.go (hunk 4 above)
Comment (Aflynn50) [reply apiserver/facades/client/modelmanager/modelmanager_test.go:518]: This is just testing an error from state, thats why I removed it.

Comment (Aflynn50) [reply apiserver/facades/client/modelmanager/modelmanager_test.go:518]: I'll add it back in and skip it instead, in case whoever touches it wants to just update it

File: apiserver/facades/client/modelmanager/modelmanager_test.go (hunk 5 above)
Comment (Aflynn50) [reply apiserver/facades/client/modelmanager/modelmanager_test.go:1267]: Pretty much, but they are more permissive because this is the state suite and we're not actually testing this stuff here

**Comment by Aflynn50** [comment]
Closed to be included in #18003 instead
